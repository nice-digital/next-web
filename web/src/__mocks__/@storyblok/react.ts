export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
