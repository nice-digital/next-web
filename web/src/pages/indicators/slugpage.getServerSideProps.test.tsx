import { render } from "@testing-library/react";

import IndicatorsDetailsPage, {
	getServerSideProps,
	IndicatorsDetailsPageProps,
} from "./[slug].page";

import type { GetServerSidePropsContext } from "next";

jest.mock("@/feeds/publications/publications", () => {
	const originalModule = jest.requireActual(
		"@/feeds/publications/publications"
	);

	//Mock the getProductDetail export
	return {
		__esModule: true,
		...originalModule,
		getProductDetail: jest.fn(() => ({
			Title: "Test title 1",
			Id: "IND1",
		})),
	};
});

const props: Required<IndicatorsDetailsPageProps> = {
	slug: "/ind-1-test-title-1",
	id: "IND1",
	product: { Title: "/test", Id: "IND1" },
};

describe("IndicatorDetailPage", () => {
	it("should match snapshot for main content", () => {
		render(<IndicatorsDetailsPage {...props} />);

		expect(document.body).toMatchSnapshot();
	});
});

describe("getGetServerSidePropsFunc", () => {
	it("should return a correct props when supplied with an id", async () => {
		const result = await getServerSideProps({
			params: { slug: "ind1-test-title-1" },
		} as unknown as GetServerSidePropsContext);

		expect(result).toStrictEqual({
			props: {
				slug: "ind1-test-title-1",
				id: "ind1",
				product: { Id: "IND1", Title: "Test title 1" },
			},
		});
	});

	describe("Redirects", () => {
		it("should return permanent redirect object URL with incorrect title", async () => {
			const redirectResult = await getServerSideProps({
				params: { slug: "ind1-incorrect-slug-title" },
			} as unknown as GetServerSidePropsContext);

			expect(redirectResult).toStrictEqual({
				redirect: {
					destination: "/indicators/ind1-test-title-1",
					permanent: true,
				},
			});
		});
		it("should return notFound if Id doesn't exist", async () => {
			const redirectResult = await getServerSideProps({
				params: { slug: "nonExistingId99-test-title-1" },
			} as unknown as GetServerSidePropsContext);

			expect(redirectResult).toStrictEqual({ notFound: true });
		});

		it("should return notFound if an unhyphenated slug is used", async () => {
			const redirectResult = await getServerSideProps({
				params: { slug: "slugwithouthyphenation" },
			} as unknown as GetServerSidePropsContext);

			expect(redirectResult).toStrictEqual({ notFound: true });
		});

		//TODO check logic for any missing cases
	});
});
