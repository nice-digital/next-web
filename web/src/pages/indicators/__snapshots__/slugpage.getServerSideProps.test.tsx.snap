// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorDetailPage should match snapshot for main content 1`] = `
<body>
  <div>
    <h1>
      /test
    </h1>
    <p>
      Indicators details page id:
      IND1
       title:
      /test
    </p>
  </div>
</body>
`;
